client/src/components/Modals/EditTeam/index.tsx

```diff
@@ -0,0 +1,195 @@
+
+  useEffect(() => {
+    fetchTeams();
+    setFormValues(teams);
+
+    const clone = teams.map((team) => ({
+      _id: team._id,
+      name: team.name,
+      color: team.color,
+      active: team.active,
+    }));
+
+    setClonedTeams([...clone]);
+    console.log(teams);
+  }, [isOpen]);
+
+  // Add a new field in form
+  const addField = () => {
+    setFormValues([
+      ...formValues,
+      { _id: 'NEW', name: '', color: '#3d4f91', active: true },
+    ]);
+  };
+
+  // Remove a field in form
+  const removeField = (i: number) => {
+    const newFormValues = [...formValues];
+    newFormValues.splice(i, 1);
+    setFormValues(newFormValues);
+  };
+
+  // Update team names (input type = text)
+  const updateFieldText = (i: number, text: string) => {
+    const formValue = formValues[i];
+    formValue.name = text;
+    setFormValues([...formValues]);
+  };
+
+  // Update team colors (input type = color)
+  const updateFieldColor = (i: number, color: string) => {
+    const formValue = formValues[i];
+    formValue.color = color;
+    setFormValues([...formValues]);
+  };
+
+  // Save newTeams and changedTeams
+  const saveForm = (): void => {
+    const newTeams = formValues.filter((team) => team._id === 'NEW');
+    const parsedNewTeams: Partial<ITeam>[] = newTeams.map(
+      ({ _id, ...rest }) => rest,
+    );
+
+    const changedTeams: ITeam[] = [];
+
+    clonedTeams.forEach((team, index) => {
+      if (
+        team.name !== formValues[index].name ||
+        team.color !== formValues[index].color
+      ) {
+        changedTeams.push(formValues[index]);
+      }
+    });
+
+    // Check if the changedTeam names is in cloned Teams
+    // Check if the new team names
+    const valuesName = formValues.map((item) => item.name);
+    const isDuplicate = valuesName.some(
+      (item, idx) => valuesName.indexOf(item) !== idx,
+    );
+    console.log(isDuplicate);
+    if (isDuplicate) {
+      Swal.fire({
+        icon: 'error',
+        title: 'Cannot create duplicate team name!',
+      });
+      return;
+    }
+
+    Swal.fire({
+      icon: 'question',
+      text: 'Are you sure you want to save your “Teams” changes?',
+      showCancelButton: true,
+      cancelButtonText: 'Cancel',
+      confirmButtonText: 'Confirm',
+    }).then((result) => {
+      if (result.isConfirmed) {
+        createManyTeams(parsedNewTeams);
+        updateManyTeams(changedTeams);
+
+        setIsOpen(false);
+        fetchTeams();
+        setFormValues([]);
+
+        Swal.fire(
+          'Team updated!',
+          'Team formation has been updated',
+          'success',
+        );
+      }
+    });
+  };
+
+  return (
+    <Modal
+      size="tiny"
+      open={isOpen}
+      onOpen={() => setIsOpen(true)}
+      onClose={() => setIsOpen(false)}
+      trigger={<TeamModalTrigger />}
+      className="team-modal"
+      {...rest}
+    >
+      <Modal.Header>Edit Teams</Modal.Header>
+      <Modal.Content>
+        <Form id="submit-team">
```

Comment:
Because you have a button at the bottom that has `type=submit` you could use the `onSubmit={saveForm}` callback for form.